model/profile.py

```diff
@@ -0,0 +1,47 @@
+from flask import Blueprint, request
+
+from mongo import User
+
+from .utils import HTTPResponse, HTTPRedirect, HTTPError, Request
+from .auth import login_required
+
+profile_api = Blueprint('profile_api', __name__)
+
+
+@profile_api.route('/', methods=['GET'])
+@profile_api.route('/<username>', methods=['GET'])
+@login_required
+def view_others_profile(user, username=None):
+    try:
+        user = user if username is None else User(username)
+        data = {
+            'username': user.username,
+            'email': user.obj.email,
+            'displayed_name': user.obj.profile.displayed_name,
+            'bio': user.obj.profile.bio
+        }
+    except:
+        return HTTPError('Profile not exist.', 404)
+
+    return HTTPResponse('Profile exist.', data=data)
+
+
+@profile_api.route('/', methods=['POST'])
+@login_required
+@Request.json(['bio'], vars_dict={'displayed_name': 'displayedName'})
+def edit_profile(user, displayed_name=None, bio=None):
```

Comment:
```suggestion
def edit_profile(user, displayed_name, bio):
```
要是請求不包含這個欄位，`Request.json` 會把他設為 `None`